### DISCUSSION TITLE
ErrorBoundaries & Suspense

### DISCUSSION THREAD

**Discussion by hazem-alabiad** [pmndrs/zustand#1882] [Q&A] [answered]
Does Zustand has built-in support for `Suspense` and `ErrorBoundary`?

**Comment by dai-shi** [comment, marked as answer]
fyi, https://twitter.com/dai_shi/status/1580931674656817153 and https://github.com/dai-shi/use-zustand.

**Reply by hazem-alabiad** [reply]
Just a tiny note: 

You need to pick the proper instance based on the `store` property value.

<img width="387" alt="image" src="https://github.com/pmndrs/zustand/assets/16942677/90247c97-7fe4-4c84-b4ee-b8c7cbfd99e7">

**Reply by dai-shi** [reply]
feel free to improve docs. cc: @dbritto-dev
